modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/portlet_url/create_url.es.js

```diff
@@ -0,0 +1,81 @@
+import {isObject, isString, isNull} from 'metal';
+
+/**
+ * Returns a portlet URL in form of a string
+ * @param {!string} basePortletURL The base portlet URL to be modified in this utility
+ * @param {object} parameters Search parameters to be added or changed in the base URL
+ * @return {string} Portlet URL
+ */
+export default function createURL(basePortletURL, parameters = {}) {
+	if (!isString(basePortletURL)) {
+		throw new TypeError('basePortletURL parameter must be a string');
+	}
+
+	if (!isObject(parameters)) {
+		throw new TypeError('parameters argument must be an object');
+	}
+
+	if (!basePortletURL.includes('http://')) {
+		basePortletURL = `http://${basePortletURL}`;
+	}
+
+	const reservedParameters = [
+		'doAsGroupId',
+		'doAsUserId',
+		'doAsUserLanguageId',
+		'p_auth',
+		'p_auth_secret',
+		'p_f_id',
+		'p_j_a_id',
+		'p_l_id',
+		'p_l_reset',
+		'p_p_auth',
+		'p_p_cacheability',
+		'p_p_i_id',
+		'p_p_id',
+		'p_p_isolated',
+		'p_p_lifecycle',
+		'p_p_mode',
+		'p_p_resource_id',
+		'p_p_state',
+		'p_p_state_rcv',
+		'p_p_static',
+		'p_p_url_type',
+		'p_p_width',
+		'p_t_lifecycle',
+		'p_v_l_s_g_id',
+		'refererGroupId',
+		'refererPlid',
+		'saveLastPath',
+		'scroll'
+	];
+
+	const portletURL = new URL(basePortletURL);
+
+	const portletURLSearchParams = new URLSearchParams(portletURL.search);
+
+	const portletID = parameters.p_p_id || portletURLSearchParams.get('p_p_id');
+
+	const namespacePrefix = Liferay.Util.getPortletNamespace(portletID);
+
+	if (isNull(portletID)) {
+		throw new TypeError('Portlet ID must not be null');
+	}
+
+	Object.keys(parameters).forEach(key => {
+		if (!isNull(reservedParameters[key])) {
+			if (!reservedParameters.includes(key)) {
```

Comment:
Remove  `.set` duplication by rewriting `if` to two steps:
1. if (!reserved) { add namespace prefix}
1. always set param